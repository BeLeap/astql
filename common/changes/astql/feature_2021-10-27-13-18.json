{
  "changes": [
    {
      "comment": "Publish astql ecosystem",
      "type": "minor",
      "packageName": "astql"
    }
  ],
  "packageName": "astql",
  "email": "vkefallinos@gmail.com"
}