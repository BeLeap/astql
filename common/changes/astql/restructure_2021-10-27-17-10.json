{
  "changes": [
    {
      "comment": "Restructure astql",
      "type": "minor",
      "packageName": "astql"
    }
  ],
  "packageName": "astql",
  "email": "vkefallinos@gmail.com"
}