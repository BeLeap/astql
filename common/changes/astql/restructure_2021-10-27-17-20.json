{
  "changes": [
    {
      "comment": "fix astql",
      "type": "minor",
      "packageName": "astql"
    }
  ],
  "packageName": "astql",
  "email": "vkefallinos@gmail.com"
}