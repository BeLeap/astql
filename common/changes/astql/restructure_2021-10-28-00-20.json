{
  "changes": [
    {
      "comment": "add features",
      "type": "minor",
      "packageName": "astql"
    }
  ],
  "packageName": "astql",
  "email": "vkefallinos@gmail.com"
}