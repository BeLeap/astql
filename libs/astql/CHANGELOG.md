# Change Log - astql

This log was last generated on Thu, 28 Oct 2021 01:08:42 GMT and should not be manually modified.

## 0.1.1
Thu, 28 Oct 2021 01:08:42 GMT

### Patches

- updates

## 0.1.0
Thu, 28 Oct 2021 00:58:13 GMT

### Minor changes

- Restructure astql
- fix astql
- add features

