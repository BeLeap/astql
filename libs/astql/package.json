{
  "name": "astql",
  "version": "0.1.2",
  "author": "Vasilis Kefallinos <vkefallinos@gmail.com>",
  "contributors": [],
  "description": "A query library for ECMAScript AST using a CSS selector like query language.",
  "main": "dist/index.js",
  "scripts": {
    "build": "digigov build --subpackages",
    "test": "digigov test src",
    "lint": "digigov lint "
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/gencodes/gencodes.git"
  },
  "bugs": "https://github.com/gencodes/gencodes/issues",
  "homepage": "https://github.com/gencodes/gencodes/",
  "keywords": [
    "ast",
    "ecmascript",
    "javascript",
    "query"
  ],
  "dependencies": {"eslint":"~8.1.0"
  },
  "devDependencies": {
    "@types/jest": "27.0.2",
    "@types/node":"~16.11.6",
    "pegjs": "~0.10.0",
    "@digigov/cli":"~0.5.24",
    "@digigov/cli-build":"0.5.24",
    "glob":"~7.2.0",
    "rimraf":"~3.0.2",
    "@digigov/cli-test":"0.5.24",
    "typescript": "4.4.4",
    "@astql/js.typescript":"~0.1.1",
    "@digigov/cli-lint":"~0.5.25"
  },
  "license": "BSD-3-Clause",
  "engines": {
    "node": ">=0.10"
  }
}
