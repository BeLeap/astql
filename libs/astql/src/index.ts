export * from './CodeParser';
export { default as ParserConfig } from './ParserConfig';
export { default as query } from './query';
export { default as defaultParserInterface } from './utils/defaultParserInterface';
export { default as multipleRequire } from './utils/multiple-require';
