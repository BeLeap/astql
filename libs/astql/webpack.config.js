const glob = require('glob');

module.exports = {
  mode: 'development',
  target: 'node', // use require() & use NodeJs CommonJS style
  entry: glob.sync('./src/**/*.{ts,tsx,js}').reduce((acc, file) => {
    acc[file.replace(/^\.\/src\//, '').replace(/\.[a-z]+$/, '')] = file;
    return acc;
  }, {}),
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        exclude: /node_modules/,
      },
    ],
  },
  resolve: {
    extensions: ['.tsx', '.ts', '.js'],
  },
  output: {
    filename: '[name].js',
    chunkFilename: '[name]-[id].js',
    path: `${__dirname  }/dist`
  },
};