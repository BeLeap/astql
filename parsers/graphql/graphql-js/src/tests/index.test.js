
import {Code} from 'astql';
import fs from 'fs';
import path from 'path';
import codeExample from './codeExample.js';

const code = new Code('file.graphql', codeExample, {
  parser: require('../index.js').default,
});
// it('code should parse without error', async () => {
//   expect(code.ast).toHaveProperty('_type');
// });
it('ast.getText() should match codeExample.txt', async () => {
  await code.parse()
  expect(code.ast.getFullText()).toMatchSnapshot();
});
it('should query generated ast', async () => {
  const result = await code.query('*')
  expect(result.length).toMatchSnapshot();
});
