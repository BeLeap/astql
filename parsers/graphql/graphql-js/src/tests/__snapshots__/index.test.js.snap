// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ast.getFullText() should match codeExample.txt 1`] = `
"# Paste or drop some GraphQL queries or schema
# definitions here and explore the syntax tree
# created by the GraphQL parser.

query GetUser($userId: ID!) {
  user(id: $userId) {
    id,
    name,
    isViewerFriend,
    profilePicture(size: 50)  {
      ...PictureFragment
    }
  }
}

fragment PictureFragment on Picture {
  uri,
  width,
  height
}
"
`;

exports[`should query generated ast 1`] = `43`;
